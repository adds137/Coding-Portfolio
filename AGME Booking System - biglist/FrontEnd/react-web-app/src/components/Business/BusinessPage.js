import React from "react"
import "../../App.scss"
import HomePageHeader from "../HomePage/HomePageHeader"
import { Link } from 'react-router-dom'
import WorkerAvailabilities from './WorkerAvailabilities'
import * as Constants from "../../../src/constants"

export default class BusinessPage extends React.Component{
    constructor(props) {
        super(props)
        this.state = {
            business: {},
            busid: this.props.match.params.id
        }
    }

    componentDidMount(){
        var apicall = Constants.BACKEND_URL + "/api/Business/findById=" + this.state.busid;
        fetch(apicall)
            .then(response => response.json())
            .then(data => {
                this.setState({
                    business: data,
                    workers: data.workers
                })
            })
        this.populatetable();
        this.populatebusinesshours()
    }

    populatetable(){
        fetch(Constants.BACKEND_URL + "/api/Business/getWorker=" + this.state.busid)
            .then(res =>{
                res.json()
                .then(data =>{
                    if(data.length > 0){
                        var input = ""
                        
                        for(var i = 0; i < data.length;i++){
                            input += "<tr>"
                            input += "<td>" + data[i].user.firstName + "</td>"
                            input += "<td>" + data[i].user.lastName + "</td>"
                            input += "<td>" + data[i].user.phoneNumber + "</td>"
                            input += "<tr>"
                        }
                        document.getElementById("workertable").innerHTML = input;
                    }
                })
            })
    }
    
    populatebusinesshours(){
        fetch(Constants.BACKEND_URL + "/api/BusinessHours/findByBusId=" + this.state.busid)
            .then(res => {
                res.json()
                .then(data => {
                    var dataArray = []
                    dataArray.push(data)
                    var daysarray = ["ERROR/TIME NOT SET","Monday","Tuesday","Wednesday","Thurday","Firday","Saturday","Sunday"]
                    var input = "<p className='has-text-weight-bold'>Business Hours</p>"

                         for(var i = 0; i < 7;i++){
                            dataArray.forEach((row) =>{
                                if(row[i].openingTime == null || row[i].closingTime == null)
                                    input += "<p>" + daysarray[i + 1] + ": CLOSED </p>"
                                else{
                                    var opening = new Date("2015-03-25T" + row[i].openingTime.toString());
                                    var closing = new Date("2015-03-25T" + row[i].closingTime.toString())
                                    console.log(row[i].openingTime)
                                    var openinghour = opening.getHours() % 12
                                    var closinghour = closing.getHours() % 12
                                    var openingmin;
                                    var closingmin;
                                    var openingmeridiem
                                    var closingmeridiem
                                    if(openinghour == 0)
                                        openinghour = 12    

                                    if(closinghour == 0)
                                        closinghour = 12

                                    if(openinghour >= 12)
                                        openingmeridiem = " pm"
                                    else
                                        openingmeridiem = " am"

                                    if(closinghour >= 12)
                                        closingmeridiem = " pm"
                                    else
                                        closingmeridiem = " am"

                                    if(opening.getMinutes() < 10)
                                        openingmin = "0" + opening.getMinutes() 
                                    else
                                        openingmin = opening.getMinutes() 

                                    if(closing.getMinutes() < 10)
                                        closingmin = "0" + closing.getMinutes()
                                    else
                                        closingmin = closing.getMinutes()

                                    var openingformatted = openinghour + ":" + openingmin + openingmeridiem
                                    var closingformatted = closinghour + ":" + closingmin + closingmeridiem
                                    input += "<p>" + daysarray[i + 1] + ": " + openingformatted +" To " + closingformatted + "</p>"
                                }               
                            });
                            document.getElementById("businesshours").innerHTML = input;
                        }
                    }
                )
            })
    }

    render(){
        return( 
            <section className="hero is-fullheight is-default is-bold">
                <div className="header">
                    <HomePageHeader/>
                </div>
                <div className="hero is-primary is-bold">
                    <h1 className="title is-1 has-text-centered py-6 business-title">{this.state.business.name}</h1>
                </div>
                <div className="hero-body">
                    <div className="container has-text-centered">
                        <div className="columns is-ventered">
                            <div className="column is-narrow">
                                <h3 className="subtitle is-4">Worker List:</h3>
                                <table className="table">
                                    <thead>
                                        <tr>
                                        <th>First Name</th>
                                        <th>Last Name</th>
                                        <th>Number</th>
                                    </tr>
                                    </thead>
                                    <tbody id="workertable"></tbody>
                                </table>
                            </div>
                            <div className="column is-6 is-offset-1">
                                <p>{this.state.business.blurb}</p>
                                <br></br>
                                <p>{this.state.business.description}</p>
                                <div className="py-6">
                                    <div className="notification is-primary">
                                        <div id="businesshours"></div>
                                    </div>
                                </div>
                                <Link to={"edit/" + this.state.business.id}>
                                    <div className="button is-primary">Edit Business</div>
                                </Link>
                            </div>
                        </div>
                        <WorkerAvailabilities busId={this.props.match.params.id} />
                    </div>
                </div>
                <div className="hero-foot">
                    <div className="container">
                        <div className="columns">
                            <div className="column">
                        <div className="notification is-link">
                            <p className="has-text-weight-bold">Address:</p>
                            <p>{this.state.business.address}</p>
                            </div>
                        </div>
                        <div className="column">
                        <div className="notification is-link">
                            <p className="has-text-weight-bold">Mobile Number:</p>
                            <p>{this.state.business.phoneNumber}</p>
                            </div>
                        </div>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}