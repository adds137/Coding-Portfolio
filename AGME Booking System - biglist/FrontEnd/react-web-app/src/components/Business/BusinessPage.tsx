import React from "react"
import "../../App.scss"
import * as Constants from "../../../src/constants"

export default class BusinessPage extends React.Component<{},any>{
    constructor(props: number) {
        super(props)
        this.state = {
            business: {},
            businessId: 1
        }
    }

    componentDidMount(){
        var apicall = Constants.BACKEND_URL + "/api/Business/findById=" + this.state.businessId
        fetch(apicall)
            .then(response => response.json())
            .then(data => {
                this.setState({
                    business: data
                })
            })
        this.populatetable();
    }

    populatetable(){
        fetch(Constants.BACKEND_URL + "/user/1")
            .then(res =>{
                res.json()
                .then(data => {
                    var dataArray = [];
                    dataArray.push(data);
                    if(dataArray.length > 0){
                        var temp :any = "";

                        dataArray.forEach((row) =>{
                            temp += "<tr>"
                            temp += "<td>" + row.firstName + "</td>"
                            temp += "<td>" + row.lastName + "</td>"
                            temp += "<td>" + row.homeAddress + "</td>"
                            temp += "<td>" + row.phoneNumber + "</td>"
                            temp += "<td>" + "TO BE ADDED" + "</td>"
                            temp += "<tr>"
                        })
                        //document.getElementById("workertable").innerHTML = temp;
                    }
                })
            })
    }
    
    render(){
        return( 
            <section className="hero is-fullheight is-default is-bold">
                <div className="hero is-primary is-bold">
                    <h1 className="title is-1 has-text-centered py-6">{this.state.business.name}</h1>
                </div>
                <div className="hero-body">
                    <div className="container has-text-centered">
                        <div className="columns is-ventered">
                            <div className="column is-narrow">
                                <h3 className="subtitle is-4">Worker List:</h3>
                                <table className="table">
                                    <thead>
                                        <tr>
                                        <th>First Name</th>
                                        <th>Last Name</th>
                                        <th>Home Address</th>
                                        <th>Number</th>
                                        <th>Email</th>  
                                    </tr>
                                    </thead>
                                    <tbody id="workertable"></tbody>
                                </table>
                            </div>
                            <div className="colum is-6 is-offset-1">
                                <p>{this.state.business.blurb}</p>
                                <br></br>
                                <p>{this.state.business.description}</p>
                            </div>
                        </div>
                    </div>
                </div>


            <div className="hero-foot">
                <div className="container">
                    <div className="columns">
                        <div className="column">
                    <div className="notification is-link">
                        <p className="has-text-weight-bold">Address:</p>
                        <p>{this.state.business.address}</p>
                        </div>
                    </div>
                    <div className="column">
                    <div className="notification is-link">
                        <p className="has-text-weight-bold">Mobile Number:</p>
                        <p>{this.state.business.phoneNumber}</p>
                        </div>
                    </div>
                    </div>
                </div>
            </div>
            </section>
        )
    }
}