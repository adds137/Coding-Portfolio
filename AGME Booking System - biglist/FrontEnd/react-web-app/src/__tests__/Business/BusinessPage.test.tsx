import React from "react"
import BusinessPage from "../../components/Business/BusinessPage";

const { shallow, mount } = require("enzyme")
const Enzyme = require("enzyme")
const Adapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new Adapter()}); 

// test if the edit businessPage component render properly
describe('<BusinessPage /> component unit test', () => {
    it('should render correctly', () =>{
        const wrapper = shallow(<BusinessPage />);
        expect(wrapper).toMatchSnapshot();
    })
})

// test if the state save the id correctly
describe('<BusinessPage /> component unit test', () => {
    const mockfn = jest.fn();
    const state = {
        businessId: 1
    };
    it("should have businessid of 1", ()=>{
        const component = shallow(<BusinessPage {... state} />)
        expect(component).toHaveLength(1);
    })
})
