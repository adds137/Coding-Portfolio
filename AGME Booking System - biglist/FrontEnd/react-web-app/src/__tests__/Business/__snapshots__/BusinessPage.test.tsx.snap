// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessPage /> component unit test should render correctly 1`] = `ShallowWrapper {}`;
